### PR TITLE
KFSPTS-21092 prevent downloading of payee ach files in dev and test

### PR DESCRIPTION
**Pull request by jhulslander**
For the unit tests to pass, the config PR: https://github.com/CU-CommunityApps/config/pull/21 will need be merged

### CODE REVIEW

File: src/test/java/edu/cornell/kfs/sys/batch/service/impl/CuBatchFileAdminAuthorizationServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
+package edu.cornell.kfs.sys.batch.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.apache.commons.lang.StringUtils;
```
Comment (cah292) [inline src/test/java/edu/cornell/kfs/sys/batch/service/impl/CuBatchFileAdminAuthorizationServiceImplTest.java:5]: Please use lang3 StringUtils instead.

File: src/test/java/edu/cornell/kfs/sys/batch/service/impl/CuBatchFileAdminAuthorizationServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    @BeforeEach
+    void setUp() throws Exception {
+        Logger.getLogger(CuBatchFileAdminAuthorizationServiceImpl.class.getName()).setLevel(Level.DEBUG);
+        cuBatchFileAdminAuthorizationServiceImpl = new CuBatchFileAdminAuthorizationServiceImpl();
+        allowedFile = buildMockBatchFile(TEST_BATCH_BASE_DIRECTORY + ALLOWED_DIRECTORY_NAME, ALLOW_FILE_NAME);
+        conditionallyPreventedFile = buildMockBatchFile(TEST_BATCH_BASE_DIRECTORY + CONDITIONALLY_PREVENT_DIRECTORY_NAME, PREVENT_FILE_NAME);
+    }
+    
+    private BatchFile buildMockBatchFile(String patch, String fileName) {
```
Comment (cah292) [inline src/test/java/edu/cornell/kfs/sys/batch/service/impl/CuBatchFileAdminAuthorizationServiceImplTest.java:33]: The first arg should be renamed from "patch" to "path"

File: src/main/java/edu/cornell/kfs/sys/batch/service/impl/CuBatchFileAdminAuthorizationServiceImpl.java [hunk 3]
```diff
@@ -40,5 +48,36 @@ public boolean canDownload(BatchFile batchFile, Person user) {
...
         return isAuthorized || super.canDownload(batchFile, user);
     }
+    
+    public boolean isDownloadOfFilePrevented(BatchFile batchFile) {
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("isDownloadOfFilePrevented, checking batch file " + batchFile.getFileName());
+            LOG.debug("isDownloadOfFilePrevented, directories that are configured to prevent download: " + preventDownloadDirectories);
+        }
+        if (StringUtils.isNotBlank(preventDownloadDirectories)) {
+            for(String individualDirectory : StringUtils.split(preventDownloadDirectories, KFSConstants.COMMA)) {
```
Comment (cah292) [inline src/main/java/edu/cornell/kfs/sys/batch/service/impl/CuBatchFileAdminAuthorizationServiceImpl.java:58]: Please add a space after the "for" keyword.

**Review by cah292 — CHANGES_REQUESTED** [review]
I found a few areas that could use minor cleanup, but other than that the changes look good.

Just to confirm, will the KFSPTS-21344 sub-task handle adding the new property to our configs across DEV/TEST/PROD? KFS might fail to boot up if this particular config property has not been specified.

File: src/test/java/edu/cornell/kfs/sys/batch/service/impl/CuBatchFileAdminAuthorizationServiceImplTest.java (hunk 1 above)
Comment (jhulslander) [reply src/test/java/edu/cornell/kfs/sys/batch/service/impl/CuBatchFileAdminAuthorizationServiceImplTest.java:5]: ok

File: src/test/java/edu/cornell/kfs/sys/batch/service/impl/CuBatchFileAdminAuthorizationServiceImplTest.java (hunk 2 above)
Comment (jhulslander) [reply src/test/java/edu/cornell/kfs/sys/batch/service/impl/CuBatchFileAdminAuthorizationServiceImplTest.java:33]: ok

**Comment by jhulslander** [comment]
I've made some changes based on the initial review .... this is ready for review again.

Sorry for not mentioning earlier, I do have a  Kuali Puppet change  PR that I am going to have Chad B or Donivan look at after this gets approved: https://github.com/CU-CommunityApps/kuali-puppet/pull/69

**Comment by cah292** [comment]
retest this please

**Review by cah292 — COMMENTED** [review]
The revised changes look good, but the unit tests are still failing. Perhaps Jenkins is using a cached version of the updated config? Or perhaps Jenkins is using its own independent config file that needs to be updated?

**Comment by jhulslander** [comment]
let me take a look

**Comment by jhulslander** [comment]
retest this please

**Comment by jhulslander** [comment]
I forgot it is a manual process to update the config repo on Jenkins .... I have done so, and the test should work now

**Comment by jhulslander** [comment]
@cah292 unit tests have successfully run now

**Review by cah292 — APPROVED** [review]
Changes look good, and unit tests pass. Merging.

**Pull request merged**